### PR TITLE
Pin upstream refs; Update headers

### PR DESCRIPTION
**Pull request by castholm**
Closes https://github.com/hexops/mach/issues/843 (I don't know if cross-repo keywords work).

- Pins upstream refs to recent-ish commits.
- Adds the missing `<GL/gl.h>` and `<GL/glx.h>`. These are not provided by OpenGL-Registry, so we steal them from Mesa instead.
- Adds EGL headers.
- Adds `<KHR/khrplatform.h>` (GLES headers depend on this).
- Removes GLES 1.0. `<GLES/1.0/gl.h>` is not the correct include path, it's supposed to be `<GLES/gl.h>` and there's not really any convenient way of including it with the Zig build system without having it conflict with the GLES 1.1 `<GLES/gl.h>`.
- Removes GLSC (the entire v1 API) for similar reasons. The docs are unclear on what the canonical include path for GLSC is (https://registry.khronos.org/OpenGL/index_sc.php just links to a `gl.h`, so maybe you are supposed to always vendor it yourself?). GLSC is both very old and very niche, so I personally doubt anyone consuming this package needs this API. GLSC2 remains. **NOTE:** When running `update.sh` you will also need to manually delete the GLSC directory because any mention of it has been removed from the script.

I also fixed a typo in the readme, replaced `stub.c` with an empty generated file and removed `linkLibC` (I don't think this is necessary since none of the headers are processed, but I could be mistaken).

Let me know if you disagree with any of the changes or want me to fix anything.

- [x] By selecting this checkbox, I agree to license my contributions to this project under the license(s) described in the LICENSE file, and I have the right to do so or have received permission to do so by an employer or client I am producing work for whom has this right.

### CODE REVIEW

File: update.sh [hunk 1]
```diff
@@ -2,14 +2,36 @@
...
+
+    initial_wd=$PWD
+    rm -rf "$dir"
+    mkdir "$dir"
+    cd "$dir"
+    git init -q
+    git fetch "$repo" "$ref" --depth 1
+    git checkout -q FETCH_HEAD
+    cd "$initial_wd"
+}
```
Comment (castholm) [inline update.sh:22]: `git clone --branch` only accepts branch and tag names, not commits :disappointed: so this is what we have to do to check out a single commit without pulling in the entire history.

File: README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # opengl-headers packaged for the Zig build system
 
-This is a Zig package which OpenGL and OpenGLES headers needed to develop and cross-compile e.g. GLFW applications.
+This is a Zig package which provides various OpenGL and OpenGL ES headers needed to develop and cross-compile e.g. GLFW applications.
```
Comment (slimsag) [inline README.md:3]:
```suggestion
This Zig package provides OpenGL and OpenGL ES headers needed to develop and cross-compile e.g. GLFW applications.
```

Comment (castholm) [reply README.md:3]: I copied the phrasing of https://github.com/hexops/linux-audio-headers

File: update.sh [hunk 3]
```diff
@@ -2,14 +2,36 @@
...
-mv _registry/api/GLES2 .
-mv _registry/api/GLES3 .
-mv _registry/api/GLSC .
-mv _registry/api/GLSC2 .
-rm -rf _registry
+OPENGL_REF=22ffb40aef5203367ad9a1431cabbba30d0230b9
+EGL_REF=b055c9b483e70ecd57b3cf7204db21f5a06f9ffe
+MESA_REF=4baea976bb585013124b91dc4e9aef793f36605a
 
+git_clone_ref() {
```
Comment (slimsag) [inline update.sh:9]:
```suggestion
# `git clone --depth 1` but at a specific revision
git_clone_ref() {
```

File: update.sh (hunk 1 above)
Comment (slimsag) [reply update.sh:22]: Nice, yeah - I use this approach in a few other places. Looks good.

File: README.md (hunk 2 above)
Comment (slimsag) [reply README.md:3]: Fair enough! I appreciate that.

**Comment by slimsag** [comment]
Thanks for this!

**Pull request merged**